invite/views.py

```diff
@@ -110,6 +110,11 @@ def amnesia(request):
             # determine user from email address.
             user = User.objects.filter(email__iexact=form.cleaned_data['email']).first()
             if user.is_active:
+                # delete previous activation codes if any, before sending a new one
+                prev_reset_links = PasswordResetInvitation.objects.filter(
+                                   email=form.cleaned_data['email'])
+                for link in prev_reset_links:
+                    link.delete()
```

Comment:
So we can actually avoid setting the variable and using the loop, as apparently all querysets have a delete method on them which deletes all members of the set. I didn't know about that until just now when I figured there was probably another way to delete multiple items. It's not something you'd want to use if your models have custom delete methods, as those won't be used, but the ones we're deleting here do not. Here's the documentation: https://docs.djangoproject.com/en/1.11/topics/db/queries/#deleting-objects
And here's what I'm suggesting (includes usage of `iexact`, which I think is probably worth using here):
`PasswordResetInvitation.objects.filter(email_iexact=form.cleaned_data['email']).delete()`